src/traits.rs

```diff
@@ -900,11 +1410,62 @@ macro_rules! prim_int_impl {
             fn pow(self, exp: u32) -> Self {
                 <$T>::pow(self, exp)
             }
+
+           fn general_reverse_bits(self, subword_bits: u32, group_subwords: u32) -> Self {
+              // Adapted from Matthew Fioravante's stdcxx-bitops, which
+              // is released under the MIT's License here:
+              // https://github.com/fmatthew5876/stdcxx-bitops
+           
+              let mut x: Self::Unsigned = self.bits_to_unsigned();
+              println!("input {:b}, as unsigned: {:b}, subword_bits: {}, group_subwords: {}", self, x, subword_bits, group_subwords);
+              let width: u32  = mem::size_of::<Self>() as u32;
+              let group_sz: u32 = width * 8 / group_subwords;
+              let k: u32 = group_sz - subword_bits;
+              println!("k {}", k);
+              {
+                  let mut up0 = |i: u32, l: usize, r: usize| {
+                      if k & i > 0 {
+                          x = ((x & (l as Self::Unsigned)) << (i as Self::Unsigned))
+                              | ((x & (r as Self::Unsigned)) >> (i as Self::Unsigned));
+                      }
+                      println!("up0({}): x = {:b}", i, x);
+                  };
+
+                  up0(1, 0x5555555555555555usize, 0xAAAAAAAAAAAAAAAAusize);
+                  up0(2, 0x3333333333333333usize, 0xCCCCCCCCCCCCCCCCusize);
+                  up0(4, 0x0F0F0F0F0F0F0F0Fusize, 0xF0F0F0F0F0F0F0F0usize);
+              }
+
+              {
+                  let mut up1 = |i: u32, s: u32, l: usize, r: usize| {
+                      if width > i && (k & s > 0) {
+                          x = ((x & (l as Self::Unsigned)) << (s as Self::Unsigned))
+                              | ((x & (r as Self::Unsigned)) >> (s as Self::Unsigned));
+                      }
+                      println!("up1({}): x = {:b}", i, x);
+                  };
+
+                  up1(1, 8, 0x00FF00FF00FF00FFusize, 0xFF00FF00FF00FF00usize);
+                  up1(2, 16, 0x0000FFFF0000FFFFusize, 0xFFFF0000FFFF0000usize);
+                  up1(4, 32, 0x00000000FFFFFFFFusize, 0xFFFFFFFF00000000usize);
```

Comment:
All of these constants need to be `u64`, since `usize` is smaller on 32-bit platforms.
(And of course the prints need to be removed before we can merge.)